### ISSUE TITLE
Report my results with default settings

### ISSUE THREAD

**Issue by bityangke** [ppengtang/pcl.pytorch#9]
I used the default settings and got the results on PASCAL VOC 2007 testset as below. Have anyone else train the network and could you share your results below? I want to locate the wrong places.
```

INFO voc_dataset_evaluator.py: 165: AP for aeroplane = 0.6428
INFO voc_dataset_evaluator.py: 165: AP for bicycle = 0.6947
INFO voc_dataset_evaluator.py: 165: AP for bird = 0.4916
INFO voc_dataset_evaluator.py: 165: AP for boat = 0.1704
INFO voc_dataset_evaluator.py: 165: AP for bottle = 0.2121
INFO voc_dataset_evaluator.py: 165: AP for bus = 0.6849
INFO voc_dataset_evaluator.py: 165: AP for car = 0.6873
INFO voc_dataset_evaluator.py: 165: AP for cat = 0.4078
INFO voc_dataset_evaluator.py: 165: AP for chair = 0.2315
INFO voc_dataset_evaluator.py: 165: AP for cow = 0.5369
INFO voc_dataset_evaluator.py: 165: AP for diningtable = 0.4312
INFO voc_dataset_evaluator.py: 165: AP for dog = 0.5664
INFO voc_dataset_evaluator.py: 165: AP for horse = 0.4256
INFO voc_dataset_evaluator.py: 165: AP for motorbike = 0.6703
INFO voc_dataset_evaluator.py: 165: AP for person = 0.0933
INFO voc_dataset_evaluator.py: 165: AP for pottedplant = 0.2565
INFO voc_dataset_evaluator.py: 165: AP for sheep = 0.4355
INFO voc_dataset_evaluator.py: 165: AP for sofa = 0.3715
INFO voc_dataset_evaluator.py: 165: AP for train = 0.6363
INFO voc_dataset_evaluator.py: 165: AP for tvmonitor = 0.5709
INFO voc_dataset_evaluator.py: 168: Mean AP = 0.4609
INFO voc_dataset_evaluator.py: 169: ~~~~~~~~
INFO voc_dataset_evaluator.py: 170: Results:
INFO voc_dataset_evaluator.py: 172: 0.643
INFO voc_dataset_evaluator.py: 172: 0.695
INFO voc_dataset_evaluator.py: 172: 0.492
INFO voc_dataset_evaluator.py: 172: 0.170
INFO voc_dataset_evaluator.py: 172: 0.212
INFO voc_dataset_evaluator.py: 172: 0.685
INFO voc_dataset_evaluator.py: 172: 0.687
INFO voc_dataset_evaluator.py: 172: 0.408
INFO voc_dataset_evaluator.py: 172: 0.232
INFO voc_dataset_evaluator.py: 172: 0.537
INFO voc_dataset_evaluator.py: 172: 0.431
INFO voc_dataset_evaluator.py: 172: 0.566
INFO voc_dataset_evaluator.py: 172: 0.426
INFO voc_dataset_evaluator.py: 172: 0.670
INFO voc_dataset_evaluator.py: 172: 0.093
INFO voc_dataset_evaluator.py: 172: 0.257
INFO voc_dataset_evaluator.py: 172: 0.436
INFO voc_dataset_evaluator.py: 172: 0.371
INFO voc_dataset_evaluator.py: 172: 0.636
INFO voc_dataset_evaluator.py: 172: 0.571
INFO voc_dataset_evaluator.py: 173: 0.461
INFO voc_dataset_evaluator.py: 174: ~~~~~~~~
INFO voc_dataset_evaluator.py: 175:
INFO voc_dataset_evaluator.py: 176: ----------------------------------------------------------
```

**Comment by bityangke** [comment]
I also use the pretrained weights provided by @ppengtang . The results seem normal.

```
INFO voc_dataset_evaluator.py: 170: Results:
INFO voc_dataset_evaluator.py: 172: 0.623
INFO voc_dataset_evaluator.py: 172: 0.693
INFO voc_dataset_evaluator.py: 172: 0.506
INFO voc_dataset_evaluator.py: 172: 0.281
INFO voc_dataset_evaluator.py: 172: 0.221
INFO voc_dataset_evaluator.py: 172: 0.718
INFO voc_dataset_evaluator.py: 172: 0.681
INFO voc_dataset_evaluator.py: 172: 0.568
INFO voc_dataset_evaluator.py: 172: 0.240
INFO voc_dataset_evaluator.py: 172: 0.613
INFO voc_dataset_evaluator.py: 172: 0.431
INFO voc_dataset_evaluator.py: 172: 0.594
INFO voc_dataset_evaluator.py: 172: 0.450
INFO voc_dataset_evaluator.py: 172: 0.662
INFO voc_dataset_evaluator.py: 172: 0.123
INFO voc_dataset_evaluator.py: 172: 0.233
INFO voc_dataset_evaluator.py: 172: 0.453
INFO voc_dataset_evaluator.py: 172: 0.520
INFO voc_dataset_evaluator.py: 172: 0.651
INFO voc_dataset_evaluator.py: 172: 0.572
INFO voc_dataset_evaluator.py: 173: 0.492
```

**Comment by ppengtang** [comment]
Hi, I'm not sure about the reason, but I can always get the same result in my computer...

There are two potential ways you can try:
1. Change the random seed. In my experiments, I found that the performance influence of different random seeds is 0.1~2%.
2. Use longer training iterations. Some other people have told me that using longer training iterations before decaying learning rate can contribute to better performance.

Hope these help you.
I would really appreciate it if you share your any findings that improve the performance.

**Comment by bityangke** [comment]
I found the iter size is not the same as caffe repo, here you use a iter size = 4 but 2 in caffe repo.

**Comment by ppengtang** [comment]
Yes, because in my experiments, iter size 4 obtains better performance than iter size 2.

**Comment by whitesockcat** [comment]
based on default settings and only got **mAP=0.4668** on PASCAL VOC 2007 testset. 

> INFO voc_dataset_evaluator.py: 165: AP for aeroplane = 0.6281
INFO voc_dataset_evaluator.py: 165: AP for bicycle = 0.6652
INFO voc_dataset_evaluator.py: 165: AP for bird = 0.4851
INFO voc_dataset_evaluator.py: 165: AP for boat = 0.1579
INFO voc_dataset_evaluator.py: 165: AP for bottle = 0.2300
INFO voc_dataset_evaluator.py: 165: AP for bus = 0.6822
INFO voc_dataset_evaluator.py: 165: AP for car = 0.6821
INFO voc_dataset_evaluator.py: 165: AP for cat = 0.4441
INFO voc_dataset_evaluator.py: 165: AP for chair = 0.2418
INFO voc_dataset_evaluator.py: 165: AP for cow = 0.5707
INFO voc_dataset_evaluator.py: 165: AP for diningtable = 0.3993
INFO voc_dataset_evaluator.py: 165: AP for dog = 0.5704
INFO voc_dataset_evaluator.py: 165: AP for horse = 0.4533
INFO voc_dataset_evaluator.py: 165: AP for motorbike = 0.6773
INFO voc_dataset_evaluator.py: 165: AP for person = 0.1326
INFO voc_dataset_evaluator.py: 165: AP for pottedplant = 0.2542
INFO voc_dataset_evaluator.py: 165: AP for sheep = 0.3992
INFO voc_dataset_evaluator.py: 165: AP for sofa = 0.4908
INFO voc_dataset_evaluator.py: 165: AP for train = 0.5755
INFO voc_dataset_evaluator.py: 165: AP for tvmonitor = 0.5963
INFO voc_dataset_evaluator.py: 168: Mean AP = 0.4668

**Comment by ppengtang** [comment]
It is very strange... I have rerun the experiments on my computer two days ago and I got the same results again...
Maybe you can try the original [Caffe implementation](https://github.com/ppengtang/oicr/tree/pcl). The Caffe implementation should be more stable. You can use the [small trick](https://github.com/ppengtang/oicr/tree/pcl#a-small-trick) during training and keep more detection boxes during testing (by increasing the "max_per_image" parameter here) to get similar results as the results reported in this PyTorch implementation. Actually, these two things have been used in this PyTorch implementation.

Hope these help you.

**Comment by bityangke** [comment]
It is really wired. Maybe it's because of the different GPU cards?

**Comment by bityangke** [comment]
I test on P40.

**Comment by bityangke** [comment]
@whitesockcat @ppengtang What GPU cards did you two use？

**Comment by whitesockcat** [comment]
@bityangke i test on k80.

**Comment by ppengtang** [comment]
@bityangke @whitesockcat I test on Titan XP. It is possible that testing on Pascal architecture based GPUs can get better performance than other GPUS...

**Comment by ppengtang** [comment]
Have you tried the Caffe implementation? Can the Caffe implementation get the reported performance on your GPUs?

**Comment by bityangke** [comment]
caffe version can get the performance.

**Comment by bityangke** [comment]
maybe single float vs. double float

**Comment by ppengtang** [comment]
Yes, I agree with you.
Thank you.

**Comment by whitesockcat** [comment]
amazing! @bityangke @ppengtang 
i test on titanxp and get **map=0.4917**.

**Comment by bityangke** [comment]
Also maybe a bug of CUDA.
 Someone said newer version pytorch may solve this problem.

**Comment by ppengtang** [comment]
Great!
Thanks!

**Comment by bityangke** [comment]
close this
